src/ansys/dpf/composites/layup_info/_layup_info.py

```diff
@@ -200,6 +201,37 @@ def get_dpf_material_id_by_analyis_ply_map(
     data_source_or_streams_provider:
         DPF data source with rst file or streams_provider. The streams provider is
         available from :attr:`.CompositeModel.core_model` (under metadata.streams_provider).
+
+    Note
+    ----
+    Cache the output because the computation can be performance critical.
+    """
+    warn(
+        "`get_dpf_material_id_by_analyis_ply_map` is deprecated. "
+        "Please use `get_dpf_material_id_by_analysis_ply_map`.",
+        category=DeprecationWarning,
+        stacklevel=2,
+    )
+    return get_dpf_material_id_by_analysis_ply_map(mesh, data_source_or_streams_provider)
+
+
+def get_dpf_material_id_by_analysis_ply_map(
+    mesh: MeshedRegion,
+    data_source_or_streams_provider: Union[DataSources, Operator],
+) -> Dict[str, np.int64]:
+    """Get Dict that maps analysis ply names to dpf_material_ids.
+
+    Parameters
+    ----------
+    mesh
+        DPF Meshed region enriched with lay-up information
+    data_source_or_streams_provider:
+        DPF data source with rst file or streams_provider. The streams provider is
+        available from :attr:`.CompositeModel.core_model` (under metadata.streams_provider).
```

Comment:
```suggestion
        available from the :attr:`.CompositeModel.core_model` attribute (under ``metadata.streams_provider``).
```
A code entity should be followed by a noun descriptor.